libs/brackets-server/embedded-ext/importfile/main.js

```diff
@@ -234,15 +239,88 @@ define(function (require, exports, module) {
         });
     }
 
+    function handleCopyFile() {
+        selectedFile = selected;
+        operation = "COPY";
+    }
+
+    function handleCutFile() {
+        selectedFile = selected;
+        operation = "CUT";
+    }
+
+    // Remove unnecessary prefix path
+    function convertRelativePath(path) {
+        var newPath = path.split("/");
+        newPath.splice(0, 2);
+        return newPath.join("/");
+    }
+
+    function handlePasteFile() {
+        selected = ProjectManager.getSelectedItem();
+        if (selected) {
+            let selectedPath = "";
+            if (selected.isFile) {
+                selectedPath = selected.parentPath;
+            } else {
+                selectedPath = selected.fullPath;
+            }
+
+            // Check incorrect copy
+            if (selectedPath.startsWith(selectedFile.fullPath)) {
```

Comment:
What if handlePasteFile() is called without assigning selectedFile (after no calling of copy or paste) ?
is it ok without validatoin fo selectedFile ?

Reply:
We can't select paste menu without selection of copy of cut.
```
+        pasteCmd.setEnabled(selected && selectedFile);
```
So I think it is okay.